bookings/appointment.go

```diff
@@ -0,0 +1,155 @@
+package bookings
+
+import (
+	"fmt"
+	// "net/url"
+	zoho "github.com/schmorrison/Zoho"
+)
+
+func (c *API) GetAppointment(request interface{}, params map[string]zoho.Parameter) (data AppointmentResponse, err error) {
+	endpoint := zoho.Endpoint{
+		Name:         GetAppointmentModule,
+		URL:          fmt.Sprintf(BookingsAPIEndpoint+"%s",GetAppointmentModule),
+		Method:       zoho.HTTPGet,
+		ResponseData: &AppointmentResponse{},
+		URLParameters: map[string]zoho.Parameter{
+			"filter_by": "",
+		},
+
+	}
+
+	for k,v := range params {
+		endpoint.URLParameters[k] = v;
+	}
+
+	err = c.Zoho.HTTPRequest(&endpoint)
+	if err != nil {
+		return AppointmentResponse{}, fmt.Errorf("Failed to retrieve appointments: %s", err)
+	}
+	if v, ok := endpoint.ResponseData.(*AppointmentResponse); ok {
+		return *v, nil
+	}
+	return AppointmentResponse{}, fmt.Errorf("Data retrieved was not 'AppointmentResponse'")
+}
+
+func (c *API) BookAppointment(request map[string]string) (data AppointmentResponse, err error) {
```

Comment:
I'm still not sure how to pass this request type to the HTTPRequest method. I know I would like to see the function signature to either include the specific parameters, or a struct that defines the specific parameters, and use that as the `endpoint.RequestBody`.

To keep backwards compatible, and not introduce undefined behaviour, after reviewing the method I am thinking defining a new `endpoint.BodyFormat`, then you can specify the required body format in the endpoint.

If this pattern seems ok to you, we can discuss the exact changes to be made to HTTPRequest method, and how your new functions should be structured.

Reply:
Yes, the pattern seems to be a good starting point.